### PR TITLE
Teach `DifferencingTransform` to `inverse_transform` with NaNs

### LINKED ISSUE #1136: [BUG] Teach `DifferencingTransform` to `inverse_transform` with NaNs
### 🐛 Bug Report

All our transforms can make `inverse_transform` on target after `make_future` except for `DifferencingTransform`. They just do nothing in this case and don't fail. We want also to not fail in this case.

### Expected behavior

Pass test:
* `test_inverse_transform_future_subset_difference_fail` (waiting for #1127), `DifferencingTransform` shouldn't touch NaNs in the target.

Fix tests on `DifferencingTransform`.

### How To Reproduce

Run `test_inverse_transform_future_subset_difference_fail` (waiting for #1127).

### Environment

_No response_

### Additional context

_No response_

### Checklist

- [X] Bug appears at the latest library version

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #1136.

## Closing issues
Closes #1136.

### CODE REVIEW

File: etna/transforms/math/differencing.py [hunk 1]
```diff
@@ -124,11 +130,8 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
         transformed = df.loc[:, pd.IndexSlice[segments, self.in_column]].copy()
         for current_segment in segments:
             to_transform = transformed.loc[:, pd.IndexSlice[current_segment, self.in_column]]
-            start_idx = to_transform.first_valid_index()
             # make a differentiation
-            transformed.loc[start_idx:, pd.IndexSlice[current_segment, self.in_column]] = to_transform.loc[
-                start_idx:
-            ].diff(periods=self.period)
+            transformed.loc[:, pd.IndexSlice[current_segment, self.in_column]] = to_transform.diff(periods=self.period)
```
Comment (alex-hse-repository) [inline etna/transforms/math/differencing.py:134]: May be be should rework the current test to check the new behaviour, I mean run tests in fixtures with NaNs

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: etna/transforms/math/differencing.py (hunk 1 above)
Comment (Mr-Geekman) [reply etna/transforms/math/differencing.py:134]: What tests do you mean?

Comment (Mr-Geekman) [reply etna/transforms/math/differencing.py:134]: Current tests for `DifferencingTransform` (`test_differencing_transform`) have a fixture with NaNs at the beginning. And tests check that everything is working.

Comment (Mr-Geekman) [reply etna/transforms/math/differencing.py:134]: Do we want to add more tests in `test_differencing_transform` or tests in `test_inference` are enough?
Ok, I'll look at it.

**Review by alex-hse-repository — APPROVED** [review]

**Pull request merged**